cpp/src/io/utilities/column_utils.cuh

```diff
@@ -59,31 +59,30 @@ rmm::device_uvector<column_device_view> create_leaf_column_device_views(
   auto leaf_columns = cudf::device_span<column_device_view>{leaf_column_views};
 
   auto iter = thrust::make_counting_iterator<size_type>(0);
-  thrust::for_each(
-    rmm::exec_policy(stream),
-    iter,
-    iter + parent_table_device_view.num_columns(),
-    [col_desc, parent_col_view = parent_table_device_view, leaf_columns] __device__(
-      size_type index) mutable {
-      col_desc[index].parent_column = parent_col_view.begin() + index;
-      column_device_view col        = parent_col_view.column(index);
-      // traverse till leaf column
-      while (cudf::is_nested(col.type())) {
-        auto const child = (col.type().id() == type_id::LIST)
-                             ? col.child(lists_column_view::child_column_index)
-                             : col.child(0);
-        // stop early if writing a byte array
-        if (col_desc[index].stats_dtype == dtype_byte_array &&
-            (child.type().id() == type_id::INT8 || child.type().id() == type_id::UINT8)) {
-          break;
-        }
-        col = child;
-      }
-      // Store leaf_column to device storage
-      column_device_view* leaf_col_ptr = leaf_columns.begin() + index;
-      *leaf_col_ptr                    = col;
-      col_desc[index].leaf_column      = leaf_col_ptr;
-    });
+  thrust::for_each(rmm::exec_policy(stream),
+                   iter,
+                   iter + parent_table_device_view.num_columns(),
+                   [col_desc, parent_col_view = parent_table_device_view, leaf_columns] __device__(
+                     size_type index) mutable {
```

Comment:
Just curious why this `mutable` is needed? I've not seen this in a device lambda.

Reply:
It is not needed, I just removed it and ran some tests without `mutable` to confirm.